include/core/mir/mutex.h

```diff
@@ -38,13 +38,6 @@ class MutexGuard
     {
     }
 
-    MutexGuard(MutexGuard&& from) noexcept
```

Comment:
Why are you dropping the explicit move-constructor here? This change doesn't stop a `MutexGuard` from being moved, but it *does* mean that you can access the contained data from a moved-from `MutexGuard`.

Reply:
Whoops, my intention was to disallow moving, as it was no longer necessary with the `.drop` method. However, I supposed it might be useful for other reasons so I'll reinstate it instead of `= delete`ing it.